### PR TITLE
Allow multiline quoted strings

### PR DESCRIPTION
**Pull request by franku**
Allow strings this: 

{
...
Plugin = "python"                                                              
      ":module_path=/python-modules"
      ":module_name=bareos-fd-local-fileset"                                       
      ":filename=/tmp/file-list"
...
}

### CODE REVIEW

File: core/src/lib/lex.cc [hunk 1]
```diff
@@ -549,6 +552,53 @@ static uint64_t scan_pint64(LEX* lf, char* str)
   return val;
 }
 
+class M {
```
Comment (arogge) [inline core/src/lib/lex.cc:555]: You might be able to find a better name for that class.

File: core/src/lib/lex.cc [hunk 2]
```diff
@@ -549,6 +552,53 @@ static uint64_t scan_pint64(LEX* lf, char* str)
...
+{
+  int i = lf->col_no;
+  while (lf->line[i]) {
+    if (lf->line[i] == '"') {
+      return true;
+    } else if (lf->line[i] != ' ') {
+      return false;
+    }
+    ++i;
+  };
```
Comment (arogge) [inline core/src/lib/lex.cc:598]: see above

File: core/src/lib/lex.cc [hunk 3]
```diff
@@ -549,6 +552,53 @@ static uint64_t scan_pint64(LEX* lf, char* str)
...
+  if (bfgets(m.pm, lf->fd) != NULL) {
+    int i = 0;
+    while (m.pm[i]) {
+      if (m.pm[i] == '"') {
+        return true;
+      } else if (m.pm[i] != ' ') {
+        return false;
+      }
+      ++i;
+    };
```
Comment (arogge) [inline core/src/lib/lex.cc:583]: I'd compare to '\0' to clarify that this is an end-of-string check.
```suggestion
    for (int i = 0; m.pm[i] != '\0'; ++i) {
      if (m.pm[i] == ' ') { continue; } /* skip spaces */
      return (m.pm[i] == '"'); /* is the first non-space a double quote? */
    }```

**Review by arogge — COMMENTED** [review]
Really nice work. Who would have thought our lexer could learn that?
Great application of RAII btw.

**Comment by franku** [comment]
Build succeeded with #8

**Pull request merged**